### PR TITLE
Improving test coverage in io.telicent.jena.abac.attributes (Core-555)

### PR DESCRIPTION
**Pull request by karolina-telicent**
Tests to cover holes in coverage in the AttributeParserEngine, AttributeParser, AttributeValue, and ValueTerm classes. I would appreciate help with the 4 tests marked with `TODO` comments, as I haven't been able to figure them out on my own so far.

### CODE REVIEW

File: rdf-abac-core/src/test/java/io/telicent/jena/abac/attributes/TestAttributeParserEngine.java [hunk 1]
```diff
@@ -0,0 +1,153 @@
...
+        AttributeSyntaxError exception = assertThrows(AttributeSyntaxError.class, () -> {
+            AttributeParserEngine aep = new AttributeParserEngine("{word");
+            AttributeExpr ae1 = aep.attributeExpression();
+        });
+        assertEquals("No RBRACE: [LBRACE:{]", exception.getMessage());
+    }
+
+    //TODO
+    // Trying to cover the lines 192-193, but it throws the "END" somewhere else
+    @Test
```
Comment (TelicentPaul) [inline rdf-abac-core/src/test/java/io/telicent/jena/abac/attributes/TestAttributeParserEngine.java:110]: That code is unreachable. readExprRel() is only ever called from one place and it's in this 

```java
if ( peek == TokenType.WORD || peek == TokenType.STRING )
            return readExprRel();
```

So the 
```java
if ( tokenizer.eof() )
```
will always be false.

File: rdf-abac-core/src/test/java/io/telicent/jena/abac/attributes/TestAttributeParserEngine.java [hunk 2]
```diff
@@ -0,0 +1,153 @@
...
+        AttributeSyntaxError exception = assertThrows(AttributeSyntaxError.class, () -> {
+            AttributeParserEngine aep = new AttributeParserEngine("{a} | ");
+            AttributeExpr ae1 = aep.attributeExpression();
+        });
+        assertEquals("END", exception.getMessage());
+    }
+
+    //TODO
+    // I can't figure out how to trigger the default case - seems to be covered by other exceptions"
+    @Test
```
Comment (TelicentPaul) [inline rdf-abac-core/src/test/java/io/telicent/jena/abac/attributes/TestAttributeParserEngine.java:119]: Again, I don't think this code is reachable. 

I think we should leave the code there though unlike the other code I've removed.

**Review by TelicentPaul — APPROVED** [review]
Added commit with some tweaks

File: rdf-abac-core/src/test/java/io/telicent/jena/abac/attributes/TestAttributeParserEngine.java (hunk 1 above)
Comment (TelicentPaul) [reply rdf-abac-core/src/test/java/io/telicent/jena/abac/attributes/TestAttributeParserEngine.java:110]: I'll remove it

**Review by rvesse — APPROVED** [review]

**Pull request merged**